src/components/modals/insertImageModal/InsertImageModal.tsx

```diff
@@ -0,0 +1,60 @@
+import { ContextModalProps } from "@mantine/modals";
+import { useCallback } from "react";
+import { AddImageUrlForm } from "./partials/AddImageUrlForm";
+import { UploadImageForm } from "./partials/UploadImageForm";
+import { Tabs } from "@mantine/core";
+
+export type TInsertImageModalProps = {
+  handleCancel?: () => void;
+  handleSubmit?: (value: string) => void;
+};
+
+export const InsertImageModal = ({
+  id,
+  context,
+  innerProps: { handleCancel, handleSubmit },
+}: ContextModalProps<TInsertImageModalProps>) => {
+  const close = useCallback(() => {
+    context.closeModal(id);
+  }, [context, id]);
+
+  const cancelHandler = useCallback(() => {
+    if (handleCancel) {
+      handleCancel();
+    }
+    close();
+  }, [close, handleCancel]);
+
+  const submitHandler = useCallback(
+    (value: string) => {
+      if (handleSubmit) {
+        handleSubmit(value);
+      }
+      close();
+    },
+    [close, handleSubmit],
+  );
+
+  return (
+    <Tabs defaultValue="upload">
+      <Tabs.List mb={"md"}>
+        <Tabs.Tab value="upload">From computer</Tabs.Tab>
+        <Tabs.Tab value="insertUrl">From URL</Tabs.Tab>
+      </Tabs.List>
```

Comment:
```suggestion
      <Tabs.List mb={"md"}>
        <Tabs.Tab value="upload" leftSection={<IconUpload size={18} />}>
          Local
        </Tabs.Tab>
        <Tabs.Tab value="insertUrl" leftSection={<IconLink size={18} />}>
          From URL
        </Tabs.Tab>
      </Tabs.List>
```

![image](https://github.com/user-attachments/assets/2ae469bd-7a00-4f13-80c8-250dbdefc78d)

I'd change the `From computer`  label and add some icons 😉 just to polish the UI